### PR TITLE
TModel substitution in Razor pages has broken intellisense

### LINKED ISSUE #3185: TModel substitution in Razor pages has broken intellisense
Mvc supports inheriting injected properties (`@inject`) from applicabe _ViewImports. A feature we support as part of this injection is the substitution of any `TModel` tokens that appear in the type name of `@inject` with the current model type. For e.g

```html
// _ViewImport
@inject CssHelper<TModel> CssHelper

// Index
@model IndexModel
```
The `CssHelper` property on Index would be of type `CssHelper<IndexModel>`. However since `TModel` isn't a real C# token, VS tooling for this is broken:

capture

Note squiggly lines under `TModel`. In addition, you do not get any auto completion for this token. Perhaps a fix for this would be to generate a `TModel` identifier in the page at design time. It would also be useful to figure out if there's a nicer way to perform this substitution.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3185

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcCSharpCodeGenerator.cs [hunk 1]
```diff
@@ -45,6 +47,22 @@ public class MvcCSharpCodeGenerator : CSharpCodeGenerator
             _injectAttribute = injectAttribute;
         }
 
+        protected override CSharpCodeWritingScope BuildClassDeclaration(CSharpCodeWriter writer)
+        {
+            if (Context.Host.DesignTimeMode &&
+                !string.IsNullOrEmpty(Context.SourceFile) &&
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcCSharpCodeGenerator.cs]: When would we not have a source file?

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs [hunk 2]
```diff
@@ -45,7 +45,7 @@ public class MvcRazorHost : RazorEngineHost, IMvcRazorHost
             new SetBaseTypeChunk
             {
                 // Microsoft.Aspnet.Mvc.Razor.RazorPage<TModel>
-                TypeName = BaseType + ChunkHelper.TModelToken,
+                TypeName =  $"{BaseType}<{ChunkHelper.TModelToken}>",
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorHost.cs]: Got an extra space after `=`

File: src/Microsoft.AspNet.Mvc.Razor.Host/Directives/ChunkHelper.cs [hunk 3]
```diff
@@ -16,7 +16,8 @@ public static class ChunkHelper
         /// Token that is replaced by the model name in <c>@inherits</c> and <c>@inject</c>
         /// chunks as part of <see cref="ChunkInheritanceUtility"/>.
         /// </summary>
-        public static readonly string TModelToken = "<TModel>";
+        public static readonly string TModelToken = "TModel";
+        private static readonly string TModelReplaceToken = $"<{TModelToken}>";
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/Directives/ChunkHelper.cs]: Wish static properties didn't have to be ordered like this :cry:

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorHostTest.cs [hunk 4]
```diff
@@ -221,6 +222,32 @@ public void BasicVisitor_GeneratesCorrectLineMappings()
...
         [Fact]
+        public void MvcRazorHost_GeneratesCorrectLineMappingsAndUsingStatementsForViewImports()
+        {
+            // Arrange
+            var fileProvider = new TestFileProvider();
+            var host = new MvcRazorHostWithNormalizedNewLine(new DefaultChunkTreeCache(fileProvider))
+            {
+                DesignTimeMode = true
+            };
+            host.NamespaceImports.Clear();
```
Comment (NTaylorMullen) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorHostTest.cs]: What's the value in clearing?

File: src/Microsoft.AspNet.Mvc.Razor.Host/MvcCSharpCodeGenerator.cs (hunk 1 above)
Comment (NTaylorMullen) [inline src/Microsoft.AspNet.Mvc.Razor.Host/MvcCSharpCodeGenerator.cs]: :boom:

**Comment by NTaylorMullen** [comment]
:shipit: once small comments are addressed.

File: test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorHostTest.cs (hunk 4 above)
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.Razor.Host.Test/MvcRazorHostTest.cs]: No idea. I just copied it from the test above.

**Pull request closed without merging**